### PR TITLE
Added KeepProjectReference flag to the tests that are not compiling.

### PR DESCRIPTION
**Pull request by venkat-raman251**
These are the tests that are not compiling as of today. I added these flags to get the build to not have errors.
@weshaggard , @ericstj  : Please have a look. 
Should we file an issue to track this?

### CODE REVIEW

**Comment by ericstj** [comment]
Can you file an issue on each these tests and reference the issue in source?
Otherwise LGTM.

**Comment by venkat-raman251** [comment]
@dotnet-bot Test Innerloop OSX Debug Build and Test

**Comment by venkat-raman251** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test

**Comment by venkat-raman251** [comment]
Issues created
#7906 #7908 #7910 #7912 #7913

**Comment by venkat-raman251** [comment]
@dotnet-bot Test Innerloop OSX Debug Build and Test

**Comment by venkat-raman251** [comment]
@dotnet-bot Test Innerloop OSX Release Build and Test

**Comment by weshaggard** [comment]
@chcosta We need to get some documentation up somewhere so folks know how to handle the tests that we mark as excluded and what it means.

File: src/System.Xml.XmlSerializer/tests/System.Xml.XmlSerializer.Tests.csproj [hunk 1]
```diff
@@ -31,11 +31,12 @@
     </ProjectReference>
       <ProjectReference Include="..\..\System.Runtime.Serialization.Primitives\src\System.Runtime.Serialization.Primitives.csproj">
       <Project>{CDF0ACB5-1361-4E48-8ECB-22E8022F5F01}</Project>
-    <Name>System.Runtime.Serialization.Primitives</Name>
+      <Name>System.Runtime.Serialization.Primitives</Name>
```
Comment (chcosta) [inline src/System.Xml.XmlSerializer/tests/System.Xml.XmlSerializer.Tests.csproj]: nit: space

**Comment by venkat-raman251** [comment]
@ericstj @weshaggard I am not sure what these OSX errors are. Could you please help me

**Comment by weshaggard** [comment]
OSX CI builds are currently in a bad state across the board. Don't block on them if the other things are done.

**Comment by bartonjs** [comment]
@venkat-raman251 Since I've merged in the fixes to X509 you would, ideally, remove the compat flag for that library.

**Comment by venkat-raman251** [comment]
@bartonjs  Removed X509

**Comment by bartonjs** [comment]
@venkat-raman251 danke :)

**Pull request merged**